### DISCUSSION TITLE
Run scanner from command line

### DISCUSSION THREAD

**Discussion by maxime-killinger** [photoview/photoview#671] [Q&A and Troubleshooting] [answered]
Hi @viktorstrate,

Since the app does not support yet Live Photo, I have to run a script to add the .MOV file to a .photoviewignore before doing a scan to keep my timeline clean. 
Is there a way to run a scan using command line ? Like that i could chain my scripts and avoid the .mov of my live photos inside my gallery. 

Thank's for the help 

Maxime

**Comment by viktorstrate** [comment, marked as answer]
It would be nice to have a proper CLI, but for the moment you could execute the HTTP request manually.
Find the request that is fired when you click the "Scan All" button and copy it as a curl command.
You should be able to start a scan by executing the copied command. Notice though that the access token will expire after a month I think.

<img width="925" alt="Screenshot 2022-04-06 at 14 30 05" src="https://user-images.githubusercontent.com/4233458/161974836-f3877eb1-5424-465d-902b-fad78e5f8a8f.png">

**Comment by maxime-killinger** [comment]
The token is saved in the database, I can change the expiration date to keep it working for more than one month. Thank's for the tips !

**Reply by Phlogi** [reply]
@maxime-killinger 
Could you provide the database command (UPDATE...) to change the token expiration, please?

**Reply by Phlogi** [reply]
It's in table **access_tokens**, column is **expire**.

**Comment by korikori** [comment]
Something that I do (migrating from PhotoPrism) to keep useless scans to a minimum is set up a cron job that only fires the scan if there are new files on the system - something like this:

```
#!/bin/bash
if [ ! -f /mnt/raid1/lockfile ]; then
  if [ "$(find /mnt/raid1/originals/some-directory /mnt/raid1/some-other-directory -cmin -10)" ]; then

	touch /mnt/raid1/lockfile
    
	curl 'https://my-domain.com/api/graphql' -X POST -H 'User-Agent: Mozilla/5.0 (
	X11; Linux x86_64; rv:106.0) Gecko/20100101 Firefox/106.0' -H 'Accept: */*' -H '
	Accept-Language: en-US,en;q=0.5' -H 'Accept-Encoding: gzip, deflate, br' -H 'Con
	tent-Type: application/json' -H 'Origin: https://my-domain.com' -H 'DNT: 1' -H
	 'Sec-Fetch-Dest: empty' -H 'Sec-Fetch-Mode: cors' -H 'Sec-Fetch-Site: same-orig
	in' -H 'Referer: https://my-domain.com/settings' -H 'Connection: keep-alive' -
	H 'Cookie: csrftoken=some-token
	dmIeL; auth-token=some-other-token' --data-raw '{"operationName":"scanAl
	lMutation","variables":{},"query":"mutation scanAllMutation {\n  scanAll {\n
	success\n    message\n    __typename\n  }\n}\n"}'
	
	rm /mnt/raid1/lockfile
  fi
fi
```

This can be further split per user, since they have separate directories and individual scan commands.

**Comment by Phlogi** [comment]
Thanks for the inspiration, this is a more generic and safe script that keep catching updates on files, wait a bit and then run the update library command once. 
We avoid missing file updates while the updater is running. 

catch_file_changes_run_once.sh
```
#!/usr/bin/env bash

MONITORDIR="/your/path/to/photos"

NOW_FILE="/tmp/catch_file_changes_now.touch"
DELAY_FILE="/tmp/catch_file_changes.touch"

AWAIT_TIME=90  # seconds
AWAIT_TIME_STR="+${AWAIT_TIME}seconds"
CMD="flock -n /tmp/catch_file_changes.lock /root/curl_cmd_run_scanner.sh"

echo "[start] $(date)"

await_and_run () {
	echo "  [${1}] waiting ${AWAIT_TIME} + 1 seconds inside await function... (eventually running your command)"
	sleep ${AWAIT_TIME}
	sleep 1  # ensure it's really newer
	touch ${NOW_FILE}
	if [[ ${NOW_FILE} -nt ${DELAY_FILE}  ]]
	then
		echo "  [${1}] awaiting succeeded (no new files added during ${AWAIT_TIME} seconds), will run command > ${CMD} <"
		eval "${CMD}"
	else
		echo "  [${1}] awaing aborted, as newer file was added, will not run command yet..."
	fi
}

# -m: Instead of exiting after receiving a single event, execute indefinitely.
# -r: Watch all subdirectories of any directories passed as arguments.  Watches will be set up recursively to an unlimited depth.  Symbolic links are not traversed.  Newly created  subdirectories will also be watched.
# -e: Event type, we use create:  A file or directory was created within a watched directory.
inotifywait -m -r -e create --format '%w%f' "${MONITORDIR}" | while read NEWFILE
do
	echo "File ${NEWFILE} has been created beneath watched folder ${MONITORDIR}"
	touch -d"${AWAIT_TIME_STR}" ${DELAY_FILE}
	# run function in background to not block detection of newer files coming in
	await_and_run "${NEWFILE}" &
	echo "do loop end of inotifywait, waiting for next newfile..."
done

# cleanup
rm -vf "${NOW_FILE}"
rm -vf "${DELAY_FILE}"

echo "[done] $(date)"
```
